requirements-release_build.txt

```diff
@@ -1,9 +1,4 @@
 build
-ml_dtypes
-numpy==2.0.0; python_version >= "3.9" and python_version <= "3.12"
```

Comment:
@justinchuby we don't need numpy during the build step?

Reply:
I don't think so. The build dependencies are all specified in pyproject.toml and will be installed there.

Reply:
does that mean that we have to specify the numpy version? because for free-threaded a certain numpy version is necessary...

Reply:
We can move numpy to the test dependency (as this PR does)

Reply:
as I understand it, I would have used dev-requirements in main, which are more flexible... and the release-requirements only in the separate release builds?

Reply:
IMO dev requirements are really only for developers in local machines. In CIs I think we should pin more versions to avoid flakyness.

Reply:
We have to define that. I think there are different possibilities. But I would agree with that. In Main, I think it might be ok if it “breaks now and then”? Then we are warned.... or we use dependabot or renovate more intensively?

Reply:
I think we should use dependabot. Having configs for repeatable CI runs (as much as possible without constraining development too much) would be preferable to me.

Reply:
yes, I would be there too... but i find dependabot somewhat limited (compared to the complexity of renovate), but we can start with that...